tests/integration/test_dependency_replacement.py

```diff
@@ -0,0 +1,74 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from os import path
+import tarfile
+
+import utils
+
+
+def test_dependency_replacement(test_env, tmpdir):
+    """
+    Check that proper versions of dependencies were used.
+
+    Process:
+    * Send new request to Cachito API to fetch retrodep with another version of dependency package
+    * Download a bundle archive
+
+    Checks:
+    * Check that the state of request is complete
+    * Check that in the response there is a key “replaces” with dict values which was replaced
+    * Check that dir deps/gomod/pkg/mod/cache/download/github.com/pkg/errors/@v/… contains
+        only the required version
+    * Check that app/go.mod file has replace directive for the specified package
+    """
+    version_replacements = test_env["dep_replacement"]["dependency_replacements"]
+    deps_values = ["name", "type", "version"]
+    dependency_replacements = [dict(zip(deps_values, item)) for item in version_replacements]
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response_created_req = client.create_new_request(
+        payload={
+            "repo": test_env["package"]["repo"],
+            "ref": test_env["package"]["ref"],
+            "pkg_managers": test_env["package"]["pkg_managers"],
+            "dependency_replacements": dependency_replacements,
+        },
+    )
+    response = client.wait_for_complete_request(response_created_req)
+    assert response.data["state"] == "complete"
+
+    replaced_dependencies = [
+        i["replaces"] for i in response.data["dependencies"] if i["replaces"] is not None
+    ]
+    names_replaced_dependencies = sorted([i["name"] for i in replaced_dependencies])
+    supposed_replaced_dependencies = sorted([i["name"] for i in dependency_replacements])
```

Comment:
I think this is a good use-case for sets. Also, you can avoid an extra iteration over the list by not storing `replaced_dependencies`
```suggestion
    names_replaced_dependencies = {
        i["replaces"]["name"] for i in response.data["dependencies"] if i["replaces"] is not None
    }
    supposed_replaced_dependencies = set([i["name"] for i in dependency_replacements])
```